### PR TITLE
Fixed set is equal to set only

### LINKED ISSUE #20089: FiniteSet not Value False
A=FiniteSet(1,2)
B=(FiniteSet(1,2),FiniteSet(1))

{1} in A
AttributeError                            Traceback (most recent call last)
<ipython-input-80-becd25ec8a42> in <module>()
----> 1 {1} in A

/usr/lib/python3/dist-packages/sympy/sets/sets.py in __contains__(self, other)
    577 
    578     def __contains__(self, other):
--> 579         symb = sympify(self.contains(other))
    580         if not (symb is S.true or symb is S.false):
    581             raise TypeError('contains did not evaluate to a bool: %r' % symb)

/usr/lib/python3/dist-packages/sympy/sets/sets.py in contains(self, other)
    305         """
    306         other = sympify(other, strict=True)
--> 307         ret = sympify(self._contains(other))
    308         if ret is None:
    309             ret = Contains(other, self, evaluate=False)

/usr/lib/python3/dist-packages/sympy/sets/sets.py in _contains(self, other)
   2001         r = false
   2002         for e in self._elements:
-> 2003             t = Eq(e, other, evaluate=True)
   2004             if isinstance(t, Eq):
   2005                 t = t.simplify()

/usr/lib/python3/dist-packages/sympy/core/relational.py in __new__(cls, lhs, rhs, **options)
    302                     return r
    303             if hasattr(rhs, '_eval_Eq'):
--> 304                 r = rhs._eval_Eq(lhs)
    305                 if r is not None:
    306                     return r

/usr/lib/python3/dist-packages/sympy/sets/sets.py in _eval_Eq(self, other)
   1899 
   1900     def _eval_Eq(self, other):
-> 1901         if not other.is_FiniteSet:
   1902             if (other.is_Union or other.is_Complement or
   1903                 other.is_Intersection or other.is_Produ

### PR DESCRIPTION
**Pull request by sidhu1012**
#### References to other Issues or PRs

Fixes #20089 

#### Brief description of what is fixed or changed
Earlier #in #subset gave wrong output due to inconsistency of #Eq with sets and expr

#### Release Notes

- sets
     - Earlier expr and sets were treated equal which gave incorrect output for some set functions(mainly : - in, is_subset), made sets and expr not to be equal

### CODE REVIEW

**Comment by sidhu1012** [comment]
Pls review

File: sympy/core/relational.py [hunk 1]
```diff
@@ -1342,6 +1343,10 @@ def is_eq(lhs, rhs):
                 return retval
 
     retval = _eval_is_eq(lhs, rhs)
+
+    if isinstance(lhs, Set) and not isinstance(rhs, Set) or not isinstance(lhs, Set) and isinstance(rhs, Set):
+        retval = False
```
Comment (oscarbenjamin) [inline sympy/core/relational.py:1348]: A better fix would be to add `_eval_Is_eq` handlers in sets/handlers. We can have a handler for `Set, Basic` that always returns False and then a handler for `Set, Set` that returns `None`.

Comment (sidhu1012) [reply sympy/core/relational.py:1348]: Yes , thats what I tried at first but the two modules are so closely linked that their circular import issue couldn't be solved.

Comment (oscarbenjamin) [reply sympy/core/relational.py:1348]: There will be some way to solve the circular import. It just sometimes involves moving some imports around in other places

Comment (sidhu1012) [reply sympy/core/relational.py:1348]: I'll give it another try

**Comment by sidhu1012** [comment]
@oscarbenjamin Review pls

File: sympy/sets/tests/test_sets.py [hunk 2]
```diff
@@ -1581,3 +1581,11 @@ def test_DisjointUnion_len():
...
     assert len(DisjointUnion(S.EmptySet, S.EmptySet, FiniteSet(x, y, z), S.EmptySet)) == 3
     raises(ValueError, lambda: len(DisjointUnion(Interval(0, 1), S.EmptySet)))
+
+def test_issue_20089():
+    B = FiniteSet(FiniteSet(1, 2), FiniteSet(1))
+    assert not 1 in B
+    assert Eq(1, FiniteSet(1, 2)) == False
+    assert FiniteSet(1) in B
+    A = FiniteSet(1, 2)
+    assert not A.issubset(B)
```
Comment (oscarbenjamin) [inline sympy/sets/tests/test_sets.py:1591]: Maybe test `A in B` as well.

File: sympy/sets/tests/test_sets.py [hunk 3]
```diff
@@ -1581,3 +1581,11 @@ def test_DisjointUnion_len():
     assert len(DisjointUnion(FiniteSet(3, 5, 7, 9), FiniteSet(x, y, z))) == 7
     assert len(DisjointUnion(S.EmptySet, S.EmptySet, FiniteSet(x, y, z), S.EmptySet)) == 3
     raises(ValueError, lambda: len(DisjointUnion(Interval(0, 1), S.EmptySet)))
+
+def test_issue_20089():
+    B = FiniteSet(FiniteSet(1, 2), FiniteSet(1))
+    assert not 1 in B
+    assert Eq(1, FiniteSet(1, 2)) == False
```
Comment (oscarbenjamin) [inline sympy/sets/tests/test_sets.py:1588]: This should be `is S.false` rather than `== False`.

File: sympy/sets/tests/test_sets.py (hunk 2 above)
Comment (oscarbenjamin) [reply sympy/sets/tests/test_sets.py:1591]: Perhaps it would be good to test an example that is a subset as well.

**Comment by oscarbenjamin** [comment]
This looks good but there could be a few more tests

**Comment by sidhu1012** [comment]
> This looks good but there could be a few more tests

I'll add more test cases

**Comment by sidhu1012** [comment]
@oscarbenjamin I have added more test cases.

**Comment by oscarbenjamin** [comment]
Looks good

**Comment by sidhu1012** [comment]
> Looks good

Can it be merged now?

**Comment by oscarbenjamin** [comment]
Yes, I was just waiting for the tests to pass

**Pull request merged**